services/datamanager/builtin/builtin.go

```diff
@@ -530,8 +530,11 @@ func getAllFilesToSync(dir string, lastModifiedMillis int) []string {
 		}
 		// If a file was modified within the past lastModifiedMillis seconds, do not sync it (data
 		// may still be being written).
-		timeSinceMod := time.Since(info.ModTime())
-		if timeSinceMod > (time.Duration(lastModifiedMillis)*time.Millisecond) || filepath.Ext(path) == datacapture.FileExt {
+		timeSinceMod := clock.Since(info.ModTime())
+		if timeSinceMod < 0 {
```

Comment:
Just curious, when does this happen?

Reply:
Only in tests, because we mock the clock that Go uses, but this is talking to the actual file system with the real time

Reply:
Added a comment!